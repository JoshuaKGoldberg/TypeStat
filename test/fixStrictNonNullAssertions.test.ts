import path from "node:path";
import { describe, expect, it } from "vitest";

import { runMutationTest } from "../src/tests/testSetup.js";

describe("strictNonNullAssertions", () => {
	it("binaryExpressions", async () => {
		const caseDir = path.join(
			import.meta.dirname,
			"cases/fixes/strictNonNullAssertions/binaryExpressions",
		);
		const { actualContent, expectedFilePath, options } =
			await runMutationTest(caseDir);
		await expect(actualContent).toMatchFileSnapshot(expectedFilePath);
		expect(options).toMatchSnapshot("options");
	});

	it("callExpressions", async () => {
		const caseDir = path.join(
			import.meta.dirname,
			"cases/fixes/strictNonNullAssertions/callExpressions",
		);
		const { actualContent, expectedFilePath, options } =
			await runMutationTest(caseDir);
		await expect(actualContent).toMatchFileSnapshot(expectedFilePath);
		expect(options).toMatchSnapshot("options");
	});

	it("objectLiterals", async () => {
		const caseDir = path.join(
			import.meta.dirname,
			"cases/fixes/strictNonNullAssertions/objectLiterals",
		);
		const { actualContent, expectedFilePath, options } =
			await runMutationTest(caseDir);
		await expect(actualContent).toMatchFileSnapshot(expectedFilePath);
		expect(options).toMatchSnapshot("options");
	});

	it("propertyAccesses", async () => {
		const caseDir = path.join(
			import.meta.dirname,
			"cases/fixes/strictNonNullAssertions/propertyAccesses",
		);
		const { actualContent, expectedFilePath, options } =
			await runMutationTest(caseDir);
		await expect(actualContent).toMatchFileSnapshot(expectedFilePath);
		expect(options).toMatchSnapshot("options");
	});

	it("returnTypes", async () => {
		const caseDir = path.join(
			import.meta.dirname,
			"cases/fixes/strictNonNullAssertions/returnTypes",
		);
		const { actualContent, expectedFilePath, options } =
			await runMutationTest(caseDir);
		await expect(actualContent).toMatchFileSnapshot(expectedFilePath);
		expect(options).toMatchSnapshot("options");
	});
});
