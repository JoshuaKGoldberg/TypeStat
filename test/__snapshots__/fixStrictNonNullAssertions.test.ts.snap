// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`strictNonNullAssertions > binaryExpressions > options 1`] = `
"[
  {
    "cleanups": {
      "suppressTypeErrors": true
    },
    "compilerOptions": {
      "strictNullChecks": true,
      "noImplicitAny": false,
      "noImplicitThis": false
    },
    "files": {
      "above": "",
      "below": "",
      "renameExtensions": false
    },
    "filters": [],
    "fixes": {
      "importExtensions": false,
      "incompleteTypes": false,
      "missingProperties": false,
      "noImplicitAny": false,
      "noImplicitThis": false,
      "noInferableTypes": false,
      "strictNonNullAssertions": true
    },
    "hints": {
      "react": {
        "propTypes": "whenRequired",
        "propTypesOptionality": "asWritten"
      }
    },
    "mutators": [
      [
        "fixImportExtensions",
        null
      ],
      [
        "fixIncompleteTypes",
        null
      ],
      [
        "fixMissingProperties",
        null
      ],
      [
        "fixNoImplicitAny",
        null
      ],
      [
        "fixNoImplicitThis",
        null
      ],
      [
        "fixNoInferableTypes",
        null
      ],
      [
        "fixStrictNonNullAssertions",
        null
      ]
    ],
    "output": {},
    "package": {
      "directory": "<rootDir>",
      "file": "<rootDir>/package.json"
    },
    "postProcess": {
      "shell": []
    },
    "projectPath": "<rootDir>/tsconfig.json",
    "types": {
      "strictNullChecks": true
    }
  }
]"
`;

exports[`strictNonNullAssertions > callExpressions > options 1`] = `
"[
  {
    "cleanups": {
      "suppressTypeErrors": true
    },
    "compilerOptions": {
      "noImplicitAny": false,
      "noImplicitThis": false,
      "strictNullChecks": true
    },
    "files": {
      "above": "",
      "below": "",
      "renameExtensions": false
    },
    "filters": [],
    "fixes": {
      "importExtensions": false,
      "incompleteTypes": false,
      "missingProperties": false,
      "noImplicitAny": false,
      "noImplicitThis": false,
      "noInferableTypes": false,
      "strictNonNullAssertions": true
    },
    "hints": {
      "react": {
        "propTypes": "whenRequired",
        "propTypesOptionality": "asWritten"
      }
    },
    "mutators": [
      [
        "fixImportExtensions",
        null
      ],
      [
        "fixIncompleteTypes",
        null
      ],
      [
        "fixMissingProperties",
        null
      ],
      [
        "fixNoImplicitAny",
        null
      ],
      [
        "fixNoImplicitThis",
        null
      ],
      [
        "fixNoInferableTypes",
        null
      ],
      [
        "fixStrictNonNullAssertions",
        null
      ]
    ],
    "output": {},
    "package": {
      "directory": "<rootDir>",
      "file": "<rootDir>/package.json"
    },
    "postProcess": {
      "shell": []
    },
    "projectPath": "<rootDir>/tsconfig.json",
    "types": {
      "strictNullChecks": true
    }
  }
]"
`;

exports[`strictNonNullAssertions > objectLiterals > options 1`] = `
"[
  {
    "cleanups": {
      "suppressTypeErrors": true
    },
    "compilerOptions": {
      "noImplicitAny": false,
      "noImplicitThis": false,
      "strictNullChecks": true
    },
    "files": {
      "above": "",
      "below": "",
      "renameExtensions": false
    },
    "filters": [],
    "fixes": {
      "importExtensions": false,
      "incompleteTypes": false,
      "missingProperties": false,
      "noImplicitAny": false,
      "noImplicitThis": false,
      "noInferableTypes": false,
      "strictNonNullAssertions": true
    },
    "hints": {
      "react": {
        "propTypes": "whenRequired",
        "propTypesOptionality": "asWritten"
      }
    },
    "mutators": [
      [
        "fixImportExtensions",
        null
      ],
      [
        "fixIncompleteTypes",
        null
      ],
      [
        "fixMissingProperties",
        null
      ],
      [
        "fixNoImplicitAny",
        null
      ],
      [
        "fixNoImplicitThis",
        null
      ],
      [
        "fixNoInferableTypes",
        null
      ],
      [
        "fixStrictNonNullAssertions",
        null
      ]
    ],
    "output": {},
    "package": {
      "directory": "<rootDir>",
      "file": "<rootDir>/package.json"
    },
    "postProcess": {
      "shell": []
    },
    "projectPath": "<rootDir>/tsconfig.json",
    "types": {
      "strictNullChecks": true
    }
  }
]"
`;

exports[`strictNonNullAssertions > propertyAccesses > options 1`] = `
"[
  {
    "cleanups": {
      "suppressTypeErrors": true
    },
    "compilerOptions": {
      "noImplicitAny": false,
      "noImplicitThis": false,
      "strictNullChecks": true
    },
    "files": {
      "above": "",
      "below": "",
      "renameExtensions": false
    },
    "filters": [],
    "fixes": {
      "importExtensions": false,
      "incompleteTypes": false,
      "missingProperties": false,
      "noImplicitAny": false,
      "noImplicitThis": false,
      "noInferableTypes": false,
      "strictNonNullAssertions": true
    },
    "hints": {
      "react": {
        "propTypes": "whenRequired",
        "propTypesOptionality": "asWritten"
      }
    },
    "mutators": [
      [
        "fixImportExtensions",
        null
      ],
      [
        "fixIncompleteTypes",
        null
      ],
      [
        "fixMissingProperties",
        null
      ],
      [
        "fixNoImplicitAny",
        null
      ],
      [
        "fixNoImplicitThis",
        null
      ],
      [
        "fixNoInferableTypes",
        null
      ],
      [
        "fixStrictNonNullAssertions",
        null
      ]
    ],
    "output": {},
    "package": {
      "directory": "<rootDir>",
      "file": "<rootDir>/package.json"
    },
    "postProcess": {
      "shell": []
    },
    "projectPath": "<rootDir>/tsconfig.json",
    "types": {
      "strictNullChecks": true
    }
  }
]"
`;

exports[`strictNonNullAssertions > returnTypes > options 1`] = `
"[
  {
    "cleanups": {
      "suppressTypeErrors": true
    },
    "compilerOptions": {
      "strictNullChecks": true,
      "noImplicitAny": false,
      "noImplicitThis": false
    },
    "files": {
      "above": "",
      "below": "",
      "renameExtensions": false
    },
    "filters": [],
    "fixes": {
      "importExtensions": false,
      "incompleteTypes": false,
      "missingProperties": false,
      "noImplicitAny": false,
      "noImplicitThis": false,
      "noInferableTypes": false,
      "strictNonNullAssertions": true
    },
    "hints": {
      "react": {
        "propTypes": "whenRequired",
        "propTypesOptionality": "asWritten"
      }
    },
    "mutators": [
      [
        "fixImportExtensions",
        null
      ],
      [
        "fixIncompleteTypes",
        null
      ],
      [
        "fixMissingProperties",
        null
      ],
      [
        "fixNoImplicitAny",
        null
      ],
      [
        "fixNoImplicitThis",
        null
      ],
      [
        "fixNoInferableTypes",
        null
      ],
      [
        "fixStrictNonNullAssertions",
        null
      ]
    ],
    "output": {},
    "package": {
      "directory": "<rootDir>",
      "file": "<rootDir>/package.json"
    },
    "postProcess": {
      "shell": []
    },
    "projectPath": "<rootDir>/tsconfig.json",
    "types": {
      "strictNullChecks": true
    }
  }
]"
`;
